### PR TITLE
[wip] brier score

### PR DESCRIPTION
**Pull request by bmosaicml**
Brier score seems of questionable usefulness. COPA results:

First number for each model is Brier score. Below we find that accuracy AND brier score both go up with model size (not good because brier score is lower = better)
```
| Category   | Benchmark   | Subtask   |   Accuracy | Number few shot   | Model                   |
|:-----------|:------------|:----------|-----------:|:------------------|:------------------------|
|            | copa        |           |   0.337194 | 0-shot            | EleutherAI/gpt-neo-125m |
|            | copa        |           |   0.63     | 0-shot            | EleutherAI/gpt-neo-125m |
|            | copa        |           |   0.381412 | 0-shot            | EleutherAI/gpt-neo-1.3B |
|            | copa        |           |   0.71     | 0-shot            | EleutherAI/gpt-neo-1.3B |
```

### CODE REVIEW

File: llmfoundry/eval/metrics/nlp.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        self.add_state('total', default=torch.tensor(0.0), dist_reduce_fx='sum')
+
+    def update(self, batch: dict, output_logits: torch.Tensor, labels: torch.Tensor):
+        probabilites = []
+        for batch_idx, cont_idx in enumerate(batch['continuation_indices']):
+            # continuation indices refer to indices in the original input's token space
+            cont_tok_logits = output_logits[batch_idx].index_select(dim=0, index=cont_idx - 1)
+            # labels have been shifted left by one index, so the cont_idx needs to be shifted as well.
+            cont_tok_targ = labels[batch_idx].index_select(dim=0, index=cont_idx - 1)
+            mean_logit_of_targ_tok = cont_tok_logits.index_select(dim=1, index=cont_tok_targ).diagonal().mean()
```
Comment (bmosaicml) [inline llmfoundry/eval/metrics/nlp.py:40]: this is the part I need some help double checking. This computes the mean output logit of the target tokens, then below we compute exp(-mean_logit). On line 45 we normalize across the C choices

@mansheej

**Comment by dakinggg** [comment]
Feel free to reopen if you are still doing this
